### PR TITLE
Changes for Bbar_s->Klnu paper (ANALYSIS-2023-03)

### PR DESCRIPTION
**Pull request by dvandyk**
Changes include:
  - updated LCSR constraint on the full set of Bbar_s->K form factors
  - three sets of LQCD constraints
  - observables for Bbar_s->Klnu
  - additions/fixed to the list of references

### CODE REVIEW

File: eos/b-decays/observables.cc [hunk 1]
```diff
@@ -355,6 +355,46 @@ namespace eos
...
+        auto imp = new Implementation<ObservableGroup>(
+            R"(Observables in $B_s\to \bar{K} \ell^-\bar\nu$ decays)",
+            R"(The option "l" selects the charged lepton flavor.)"
+            R"(The option "form-factors" selects the form factor parametrization.)",
+            {
+                make_observable("B_s->Klnu::dBR/dq2", R"(d\mathcal{B}(B_s\to \bar{K}\ell^-\bar\nu)/dq^2)",
+                        Unit::InverseGeV2(),
+                        &BToPseudoscalarLeptonNeutrino::differential_branching_ratio,
+                        std::make_tuple("q2"),
+                        Options{ { "U", "u" }, {"q", "s"}, { "I", "1/2" } }),
```
Comment (mreboud) [inline eos/b-decays/observables.cc:372]: hmm why is the option "U" enforced here?

File: eos/constraints/B-to-P-form-factors.yaml [hunk 2]
```diff
@@ -887,6 +887,295 @@ B_s->D_s::f_++f_0@HPQCD:2019A:
 # B_s -> K form factors #
 #########################
 # {{{
+B_s->K::form-factors[f_+,f_0,f_T]@BvDV:2023A:
```
Comment (mreboud) [inline eos/constraints/B-to-P-form-factors.yaml:890]: Can you use a more compact and readable syntax? e.g.
```
kinematics:
  [ {q2: -10.0}, {q2: -5.0}, {q2: 0.0}, {q2: 5.0},
    {q2: -10.0}, {q2: -5.0}, {q2: 5.0},
    {q2: -10.0}, {q2: -5.0}, {q2: 0.0}, {q2: 5.0} ]
```

File: eos/constraints/B-to-P-form-factors.yaml [hunk 3]
```diff
@@ -887,6 +887,295 @@ B_s->D_s::f_++f_0@HPQCD:2019A:
...
+    - 0.0005555378959649752
+    - 0.00021766685503578996
+    - 0.0003766751242088745
+    - 0.0005201042743409845
+    - 0.0012028645352778738
+  dof: 11
+B_s->K::f_++f_0@RBC+UKQCD:2023A:
+  type: MultivariateGaussian(Covariance)
+  observables:
+  - B_s->K::f_+(q2)
```
Comment (mreboud) [inline eos/constraints/B-to-P-form-factors.yaml:1066]: Ditto

File: eos/constraints/B-to-P-form-factors.yaml [hunk 4]
```diff
@@ -887,6 +887,295 @@ B_s->D_s::f_++f_0@HPQCD:2019A:
...
+    - 0.00046198808652016677
+    - 0.0006053546394420631
+    - 0.0002690610999385106
+    - 0.0003736797936133099
+    - 0.0005555378959649752
+    - 0.00021766685503578996
+    - 0.0003766751242088745
+    - 0.0005201042743409845
+    - 0.0012028645352778738
+  dof: 11
```
Comment (mreboud) [inline eos/constraints/B-to-P-form-factors.yaml:1062]: `dof` is not needed anymore

File: eos/constraints/B-to-P-form-factors.yaml [hunk 5]
```diff
@@ -887,6 +887,295 @@ B_s->D_s::f_++f_0@HPQCD:2019A:
...
+B_s->K::f_++f_0@RBC+UKQCD:2023A:
+  type: MultivariateGaussian(Covariance)
+  observables:
+  - B_s->K::f_+(q2)
+  - B_s->K::f_+(q2)
+  - B_s->K::f_0(q2)
+  - B_s->K::f_0(q2)
+  - B_s->K::f_0(q2)
+  kinematics:
+  - q2: 17.6000
```
Comment (mreboud) [inline eos/constraints/B-to-P-form-factors.yaml:1072]: Why do you use 4 digits precision?

File: eos/form-factors/analytic-b-to-psd-dkmmo2008-impl.hh [hunk 6]
```diff
@@ -197,6 +199,7 @@ namespace eos
 
         // hadronic parameters
         UsedParameter MB;
+        UsedParameter fB; // Added by Carolina for cross checks
```
Comment (mreboud) [inline eos/form-factors/analytic-b-to-psd-dkmmo2008-impl.hh:202]: Fix comment and usage?

File: eos/form-factors/analytic-b-to-psd-dkmmo2008-impl.hh [hunk 7]
```diff
@@ -321,6 +326,8 @@ namespace eos
         std::function<double (const double &)> rescale_factor_p;
         std::function<double (const double &)> rescale_factor_0;
         std::function<double (const double &)> rescale_factor_T;
+        SpecifiedOption opt_decay_constant;
```
Comment (mreboud) [inline eos/form-factors/analytic-b-to-psd-dkmmo2008-impl.hh:329]: I understand the usage, but can it be documented for the future?

File: eos/form-factors/mesonic-processes.hh [hunk 8]
```diff
@@ -79,6 +79,15 @@ namespace eos
         static constexpr const double mR2_0p = 5.540 * 5.540; // B_{u,d} scalar: M(B_s scalar) - M(B_s^*) + M(B_{u,d}^*)
         static constexpr const double tp = (m_B + m_P) * (m_B + m_P);
         static constexpr const double tm = (m_B - m_P) * (m_B - m_P);
+        // zero of the conformal mapping: z(t0, t0) = 0.0
+        // This optimal value follows from z(0, t0) = - z(tm, t0)
+        static constexpr double t0 = 14.7;
```
Comment (mreboud) [inline eos/form-factors/mesonic-processes.hh:84]: Why are these values hard-coded? They are supposed to be parameters in the BFW parameterization

File: eos/parameters/hadronic-matrix-elements.yaml [hunk 9]
```diff
@@ -597,6 +597,29 @@ groups:
...
+          max:      14.7
+          latex:   '$t_0^{B_s\to K}$'
+          unit:     'GeV'
+      'B_s->K::tp@BFW2010' :
+          central:  34.346398
+          min:      34.346398
+          max:      34.346398
+          latex:    '$t_+^{B_s\to K^*}$'
+          comment:  '(mBs + mK)^2'
+          unit:     'GeV'
```
Comment (mreboud) [inline eos/parameters/hadronic-matrix-elements.yaml:622]: The unit is wrong, it should be GeV^2

**Review by mreboud — CHANGES_REQUESTED** [review]
Thanks for the work, I added a couple of comments.

File: eos/b-decays/observables.cc (hunk 1 above)
Comment (dvandyk) [reply eos/b-decays/observables.cc:372]: Because it is a ``b->ulnu`` transition.

File: eos/b-decays/observables.cc [hunk 10]
```diff
@@ -355,6 +355,46 @@ namespace eos
...
+    // {{{
+    ObservableGroup
+    make_bs_to_k_l_nu_group()
+    {
+        auto imp = new Implementation<ObservableGroup>(
+            R"(Observables in $B_s\to \bar{K} \ell^-\bar\nu$ decays)",
+            R"(The option "l" selects the charged lepton flavor.)"
+            R"(The option "form-factors" selects the form factor parametrization.)",
+            {
+                make_observable("B_s->Klnu::dBR/dq2", R"(d\mathcal{B}(B_s\to \bar{K}\ell^-\bar\nu)/dq^2)",
```
Comment (dvandyk) [inline eos/b-decays/observables.cc:368]: The LaTeX description is incorrect. It should be ``\bar{B}_s\to K\ell^-\bar\nu``.

File: eos/b-decays/observables.cc [hunk 11]
```diff
@@ -355,6 +355,46 @@ namespace eos
...
+                        Unit::InverseGeV2(),
+                        &BToPseudoscalarLeptonNeutrino::differential_branching_ratio,
+                        std::make_tuple("q2"),
+                        Options{ { "U", "u" }, {"q", "s"}, { "I", "1/2" } }),
+
+                make_observable("B_s->Klnu::BR", R"(\mathcal{B}(B_s\to \bar{K}\ell^-\bar\nu))",
```
Comment (dvandyk) [inline eos/b-decays/observables.cc:374]: Ditto on the LaTeX description.

File: eos/b-decays/observables.cc (hunk 1 above)
Comment (mreboud) [reply eos/b-decays/observables.cc:372]: Ah of course, sorry, I got confused

File: eos/constraints/B-to-P-form-factors.yaml (hunk 2 above)
Comment (dvandyk) [reply eos/constraints/B-to-P-form-factors.yaml:890]: Lovely, requires merging #674 first.

Comment (mreboud) [reply eos/constraints/B-to-P-form-factors.yaml:890]: No no, absolutely not. This is a pure Yaml thing, look at the other constraints

File: eos/form-factors/analytic-b-to-psd-dkmmo2008-impl.hh (hunk 7 above)
Comment (dvandyk) [reply eos/form-factors/analytic-b-to-psd-dkmmo2008-impl.hh:329]: Yes, a comment here would be approriate.

File: eos/parameters/hadronic-matrix-elements.yaml (hunk 9 above)
Comment (dvandyk) [reply eos/parameters/hadronic-matrix-elements.yaml:622]: Fixed locally.

File: eos/form-factors/analytic-b-to-psd-dkmmo2008-impl.hh (hunk 6 above)
Comment (dvandyk) [reply eos/form-factors/analytic-b-to-psd-dkmmo2008-impl.hh:202]: And we thought we had fixed all of them! Fixed locally.

File: eos/constraints/B-to-P-form-factors.yaml (hunk 2 above)
Comment (dvandyk) [reply eos/constraints/B-to-P-form-factors.yaml:890]: Fixed locally, for all added constraints and the renamed RBC/UKQCD constraint. We need to automatize this. Last time I failed miserably, due to floats not being stable under serialization... :-/

File: eos/constraints/B-to-P-form-factors.yaml (hunk 5 above)
Comment (dvandyk) [reply eos/constraints/B-to-P-form-factors.yaml:1072]: Fixed.

File: eos/form-factors/analytic-b-to-psd-dkmmo2008-impl.hh (hunk 7 above)
Comment (dvandyk) [reply eos/form-factors/analytic-b-to-psd-dkmmo2008-impl.hh:329]: Fixed.

File: eos/form-factors/mesonic-processes.hh (hunk 8 above)
Comment (dvandyk) [reply eos/form-factors/mesonic-processes.hh:84]: I agree, I overlooked it.

Comment (dvandyk) [reply eos/form-factors/mesonic-processes.hh:84]: These value are added in addition to the BFW parametrization. Does not hurt at all, the parameters still exist and are used!

Comment (mreboud) [reply eos/form-factors/mesonic-processes.hh:84]: But t0 is not and should not be used here, right?

**Pull request merged**